thetis/coordsys.py

```diff
@@ -0,0 +1,97 @@
+"""
+Generic methods for converting data between different spatial coordinate systems.
+Uses pyproj library.
+"""
+
+import pyproj
+import numpy as np
+
+# NOTE this adds pyproj as dependencency
+# TODO generalize; define unique names for commonly-used systems
+
+
+UTM_ZONE10 = pyproj.Proj(
+    proj='utm',
+    zone=10,
+    datum='NAD83',
+    units='m',
+    errcheck=True)
+SPCS_N_OR = pyproj.Proj(init='nad27:3601', errcheck=True)
+LL_WGS84 = pyproj.Proj(proj='latlong', datum='WGS84', errcheck=True)
+LL_WO = pyproj.Proj(proj='latlong', nadgrids='WO', errcheck=True)  # HARN
+
+
+def convertCoords(x, y, fromSys, toSys):
+    """Converts x,y from fromSys to toSys."""
+    if isinstance(x, np.ndarray):
+        # proj may give wrong results if nans in the arrays
+        lon = np.ones_like(x) * np.nan
+        lat = np.ones_like(y) * np.nan
+        goodIx = np.logical_and(np.isfinite(x), np.isfinite(y))
+        lon[goodIx], lat[goodIx] = pyproj.transform(
+            fromSys, toSys, x[goodIx], y[goodIx])
+    else:
+        lon, lat = pyproj.transform(fromSys, toSys, x, y)
+    return lon, lat
+
+
+def getVectorRotationMatrix(lon, lat, target_csys, source_csys=None,
```

Comment:
lower_case ?

Does this work for all coordinate transformations - is there some assumption about orthogonality here? I guess lon, lat velocities are kind of special (in that they're not actually vectors in the lon, lat coordinate space), but I _think_ this still does the right thing?

lon, lat should probably be more generic x,y? Also from the __doc__ it's not quite clear whether the point itself is specified in target_csys or source_csys. Finally we have fromSys and toSys in the previous routine, and target_csys and source_csys here.

Reply:
Thanks. Updated function name, signature and docstrings. Yes, as you mentioned below this is only used to rotate vectors defined in lat/lon components to local x/y directions.